### PR TITLE
Bug fixes and improvements

### LINKED ISSUE #3163: [Bug]: Invalid sync notification is dead code
### Describe the bug

<https://github.com/YimMenu/YimMenu/commit/ce1d22a23882b40410837658b959a47b273186f7>
<https://github.com/YimMenu/YimMenu/blame/54c7dada828384b96c4bffd3a0d8fd87c98a017d/src/core/settings.hpp#L171>

### Steps To Reproduce

.

### Expected Behavior

.

### Logs

.

### Screenshots / Video

.

### Store

Steam

### Language

English

### YimMenu Language

English (US)

### Additional context

.

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

### PR DESCRIPTION
**Pull request by Mr-X-GTA**
- Fixed crashes in the script VM that occurred during transitions or loading/unloading
- Fixed game crash when using Invite Player
- Fixed Turn Into Beast
- Improved exception handler
- Removed Breakup Kick > Show Cheating Message (patched)
- Removed some unused settings

Closes https://github.com/YimMenu/YimMenu/issues/3163
Closes https://github.com/YimMenu/YimMenu/issues/2887

### CODE REVIEW

**Comment by gir489returns** [comment]
I guess this also fixes #2887 but I think that the code for it was pushed in a giant MBG patch as well. So we can tag it as "fixed" with this PR too.

**Review by gir489returns — APPROVED** [review]

**Comment by snowleopardd** [comment]
For some reason i'm getting false positives with this build on windows defender

> Download the artifacts for this pull request:
> 
> * [binary.zip](https://nightly.link/YimMenu/YimMenu/actions/artifacts/1684186308.zip)
> * [pdb.zip](https://nightly.link/YimMenu/YimMenu/actions/artifacts/1684186310.zip)

**Comment by gir489returns** [comment]
IDK why your Windows Defender is saying that, mine doesn't.

https://www.virustotal.com/gui/file/a9109dc1e02ed81961ea5ff9e64cc6c7e3e7bcf1885b928265704e2506398959

https://www.virustotal.com/gui/file/e10ef81a37a2d5b828958b069a1d8a16d9c2e70c5ea5b2456cc6a1bac562b35c

This is the binary from this PR and #3322. What is weird, is 3322's binary is the one being flagged by Microsoft as Program:Win32/Wacapew.C!ml. I really don't know what it's detecting.

**Comment by snowleopardd** [comment]
> IDK why your Windows Defender is saying that, mine doesn't.
> 
> https://www.virustotal.com/gui/file/a9109dc1e02ed81961ea5ff9e64cc6c7e3e7bcf1885b928265704e2506398959
> 
> https://www.virustotal.com/gui/file/e10ef81a37a2d5b828958b069a1d8a16d9c2e70c5ea5b2456cc6a1bac562b35c
> 
> This is the binary from this PR and #3322. What is weird, is 3322's binary is the one being flagged by Microsoft as Program:Win32/Wacapew.C!ml. I really don't know what it's detecting.

image

Maybe it's because i updated my windows today, but still strange because 64faa40 doesn't get detected by my windows defender, but this one does

**Comment by gir489returns** [comment]
I mean, if you feel our build environment is injecting malware, feel free to build it yourself. Or better yet, prove where the malware is in the DLL. Just saying "DETECTION.... LE BAD!" on a video game cheat isn't helpful.

**Comment by snowleopardd** [comment]
> I mean, if you feel our build environment is injecting malware, feel free to build it yourself. Or better yet, prove where the malware is in the DLL. Just saying "DETECTION.... LE BAD!" on a video game cheat isn't helpful.

I never said it's malware, just wanted to point it out because it's never worth it having your binaries flagged by windows defender if the issue can be prevented, otherwise this could lead to dumb people being concerned and creating more unnecessary issues in the repo.

File: src/core/scr_globals.hpp [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace big::scr_globals
 	static inline const script_global gsbd_fm(1845281);
 	static inline const script_global gsbd_kicking(1877042);
 	static inline const script_global gsbd_fm_events(1916617);
-	static inline const script_global gsbd_block_c(2657971);
+	static inline const script_global gsbd_block_c(2652592);
```
Comment (lonelybud) [inline src/core/scr_globals.hpp:11]: `gsbd_block_c` looks duplicate of `globalplayer_bd`. Infact i dont see where we are using this global `gsbd_block_c`

File: src/gui/components/components.hpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ namespace big
 		template<template_str cmd_str, ImVec2 size = ImVec2(0, 0), ImVec4 color = ImVec4(0.24f, 0.23f, 0.29f, 1.00f)>
 		static void player_command_button(player_ptr player = g_player_service->get_selected(), const std::vector<uint64_t> args = {}, std::optional<const std::string_view> label_override = std::nullopt)
 		{
-			static player_command* command = dynamic_cast<player_command*>(command::get(rage::joaat(cmd_str.value)));
+			static player_command* command = player_command::get(rage::joaat(cmd_str.value));
```
Comment (lonelybud) [inline src/gui/components/components.hpp:60]: how it got compiled. It should show error`cannot convert from 'big::command *' to 'big::player_command *'`

File: src/core/scr_globals.hpp (hunk 1 above)
Comment (Mr-X-GTA) [reply src/core/scr_globals.hpp:11]: No, it's not a duplicate

File: src/gui/components/components.hpp (hunk 2 above)
Comment (Mr-X-GTA) [reply src/gui/components/components.hpp:60]: No, it returns a player_command*

**Comment by ghost** [comment]
Why could this crash have happened? I just don't know how to read the cout file, sorry.
[cout.log](https://github.com/user-attachments/files/16157161/cout.log)

File: src/gui/components/components.hpp (hunk 2 above)
Comment (lonelybud) [reply src/gui/components/components.hpp:60]: correct.

File: src/core/scr_globals.hpp (hunk 1 above)
Comment (lonelybud) [reply src/core/scr_globals.hpp:11]: LGTM

File: src/gui/components/components.hpp (hunk 2 above)
Comment (lonelybud) [reply src/gui/components/components.hpp:60]: LGTM

**Comment by Yimura** [comment]
@gir489returns, I think it's flagging us for using GetAsyncKeyState. It happens often with binaries that import Windows kernel32 functions that malware usually abuse, neither is the binary signed or known in any DB (since it's that new) as such it just outright flags the file.

**Review by Yimura — APPROVED** [review]

**Pull request merged**